### PR TITLE
Adds ObjectId type for the mongo connector

### PR DESCRIPTION
**Pull request by 0x4139**
I've added a new function that converts the obtained object id to it's string representation

### CODE REVIEW

File: types.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ const (
 	// Variable length character data.
 	VarChar = "varchar"
 
+	ObjectId = "ObjectId"
```
Comment (scritchley) [inline types.go:19]: Please add a godoc comment to this.

**Review by scritchley — CHANGES_REQUESTED** [review]

File: conn.go [hunk 2]
```diff
@@ -388,6 +390,22 @@ var timestampConverter = valueConverterFunc(func(val interface{}) (driver.Value,
...
+// objectIdConverter converts a value from the underlying json response into a objectid hex string.
+var objectIdConverter = valueConverterFunc(func(val interface{}) (driver.Value, error) {
+	if val == nil {
+		return nil, nil
+	}
+	if vv, ok := val.(string); ok {
+		data, err := base64.StdEncoding.DecodeString(vv)
+		if err != nil {
+			return nil, fmt.Errorf("%s: failed to convert %v into type objectId hex string", DriverName, val)
+		} else {
```
Comment (scritchley) [inline conn.go:402]: This else could be removed.

File: conn.go [hunk 3]
```diff
@@ -388,6 +390,22 @@ var timestampConverter = valueConverterFunc(func(val interface{}) (driver.Value,
...
+			return fmt.Sprintf("%x", data), nil
```
Comment (scritchley) [inline conn.go:403]: hex.EncodeToString(data) could be used here instead (it might be slightly faster)

File: conn.go [hunk 4]
```diff
@@ -388,6 +390,22 @@ var timestampConverter = valueConverterFunc(func(val interface{}) (driver.Value,
 	return nil, fmt.Errorf("%s: failed to convert %v (%T) into type time.Time", DriverName, val, val)
 })
 
+// objectIdConverter converts a value from the underlying json response into a objectid hex string.
+var objectIdConverter = valueConverterFunc(func(val interface{}) (driver.Value, error) {
```
Comment (scritchley) [inline conn.go:394]: Please add a test for this.

**Review by scritchley — CHANGES_REQUESTED** [review]

**Comment by ryanslade** [comment]
Thanks, but we've decided to deprecate this library in favour of https://github.com/prestodb/presto-go-client

**Pull request closed without merging**